README.md

```diff
@@ -7,17 +7,147 @@
 
-Have students practice creating issues on a repository (it could even be _this_ repository)
+![mean girls](http://i.giphy.com/4CP58gxwbBy2Q.gif)
 
-For the tests, redefine a global fetch as a spy and make sure that the student's functions are calling the right endpoint(s)
+Getting data from the Github API with `fetch` is super simple. If we're
+just trying to `GET` some JSON, we can do this:
+
+```js
+fetch('https://api.github.com/repos/jquery/jquery/commits').
+  then(resp => resp.json() ).
+  then(json => console.log(json))
+```
+
+Keeping in mind that we can use the `json` method of the `Body` mixin to
+render our response as JSON, and that each `then` passes its return
+value to the next `then` as an argument.
+
+### Authentication Token
+
+https://developer.github.com/v3/oauth/#3-use-the-access-token-to-access-the-api
+
+Github's API uses [OAuth2](https://developer.github.com/v3/oauth/) for
+authorization. Setting up the full OAuth2 authorization code grant workflow is beyond
+the scope of this lab, but it is described well in the Github
+[docs](https://developer.github.com/v3/oauth/), and I highly recommend
+you give it a read.
+
+Fortunately, Github also allows you to generate your own personal
+authorization token that we can use to give us authorized access to the
+API.
+
+If you already have a personal token that you've been using to make API
+requests, you can keep using that one. Otherwise, you'll need to
+generate a new one.
+
+To start, go to https://github.com/settings/tokens and click "Generate
+new token." Name it something like "Learn.co", and check `repo` scope. Once you generate
+the token, make sure to copy and paste it somewhere, because once you
+leave that page, you won't be able to see it again.
+
+Using the token to [access the API](https://developer.github.com/v3/oauth/#3-use-the-access-token-to-access-the-api) is a simple matter of creating an `Authorization` header with our request.
+
+We need to provide our authorization token in order to list our own
+repositories with this API, so let's add our `Authorization` header
+(don't forget to assign your token to `const token`).
+
+```js
+const token = "YOUR_TOKEN_HERE"
+fetch(`https://api.github.com/user/repos`, {
+  headers: {
+    Authorization: `token ${token}`
+  }
+}).then(res => res.json()).then(json => console.log(json))
+```
+
+We just pass the desired headers as part of a second `options` argument to
+`fetch` and we are in business. Easy as that!
+
+### Getting Past GET
+
+While `GET` operations are straightforward, when we're building out full
+applications, we often need to use other HTTP verbs, such as `POST`, to
+write data as well as read it. Luckily, it's very easy to `POST` with
+`fetch` as well.
+
+Let's look at an example of posting a new comment to a commit with the
+Github API. Replace the commit with a commit from one of your
+repositories, and use your token if you want to try this out.
+
+```js
+const token = "YOUR_TOKEN_HERE"
```

Comment:
Would it be too much to have students put their token in a `.env` file (which we can add to the .gitignore on our end) and use `dotenv()` to load it? https://www.npmjs.com/package/dotenv